cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultPreviewRequestQueue.java

```diff
@@ -99,11 +99,12 @@ public DefaultPreviewRequestQueue(CConfiguration cConf, PreviewStore previewStor
 
   @Override
   public void add(PreviewRequest previewRequest) {
+    previewStore.add(previewRequest.getProgram().getParent(), previewRequest.getAppRequest());
```

Comment:
I assume this is the persisted queue?? Where will this entry getting removed?

Reply:
Entry is persisted before adding to the in-memory queue. If in-memory queue is full with capacity, this entry is removed on line `104`.

Reply:
How about on dequeue? Does it get removed?

Reply:
yes on dequeue we call 
```
previewStore.setPreviewRequestPollerInfo(previewRequest.getProgram().getParent(), pollerInfo);
```

This removes from WAITING state.